File: 백준/Silver/11047. 동전 0/동전 0.py
import sys
readline = sys.stdin.readline
n, k = map(int, readline().split())
coins = [int(readline()) for _ in range(n)]

ans, idx = 0, n
while k > 0:
    idx -= 1
    ans += k // coins[idx]
    k %= coins[idx]
print(ans)

File: 백준/Silver/11047. 동전 0/동전 0.cc
#include <iostream>
using namespace std;

int main() {
    ios::sync_with_stdio(0); cin.tie(0);
    int N, K; cin >> N >> K;
    int coins[N];
    for(int i=0; i<N; i++) cin >> coins[i];
    
    int ans = 0, idx = N-1;
    while(K) {
        if(coins[idx] <= K) {
            ans += K / coins[idx];
            K %= coins[idx];
        }
        idx--;
    }
    cout << ans;
}